src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpGoToImplementation.cs

```diff
@@ -0,0 +1,152 @@
+        {
+            var project = ProjectName;
+            await TestServices.SolutionExplorer.AddFileAsync(project, "FileImplementation.cs", cancellationToken: HangMitigatingCancellationToken);
+            await TestServices.SolutionExplorer.OpenFileAsync(project, "FileImplementation.cs", HangMitigatingCancellationToken);
+            await TestServices.Editor.SetTextAsync(
+@"class Implementation : IBar
+{
+}
+", HangMitigatingCancellationToken);
+            await TestServices.SolutionExplorer.CloseCodeFileAsync(project, "FileImplementation.cs", saveFile: true, HangMitigatingCancellationToken);
+            await TestServices.SolutionExplorer.AddFileAsync(project, "FileInterface.cs", cancellationToken: HangMitigatingCancellationToken);
+            await TestServices.SolutionExplorer.OpenFileAsync(project, "FileInterface.cs", HangMitigatingCancellationToken);
+            await TestServices.Editor.SetTextAsync(
+@"interface IBar
+{
+}", HangMitigatingCancellationToken);
+            await TestServices.Editor.PlaceCaretAsync("interface IBar", charsOffset: 0, HangMitigatingCancellationToken);
+            await TestServices.Editor.GoToImplementationAsync(HangMitigatingCancellationToken);
+            Assert.Equal("FileImplementation.cs", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
+            await TestServices.EditorVerifier.TextContainsAsync(@"class Implementation$$", assertCaretPosition: true, HangMitigatingCancellationToken);
+            Assert.True(await TestServices.Shell.IsActiveTabProvisionalAsync(HangMitigatingCancellationToken));
+        }
+
+        [IdeFact]
+        public async Task GoToImplementationFromMetadataAsSource()
+        {
+            var project = ProjectName;
+            await TestServices.SolutionExplorer.AddFileAsync(project, "FileImplementation.cs", cancellationToken: HangMitigatingCancellationToken);
+            await TestServices.SolutionExplorer.OpenFileAsync(project, "FileImplementation.cs", HangMitigatingCancellationToken);
+            await TestServices.Editor.SetTextAsync(
+@"using System;
+
+class Implementation : IDisposable
+{
+    public void SomeMethod()
+    {
+        IDisposable d;
+    }
+}", HangMitigatingCancellationToken);
+            await TestServices.Editor.PlaceCaretAsync("IDisposable d", charsOffset: -1, HangMitigatingCancellationToken);
+            await TestServices.Editor.GoToDefinitionAsync(HangMitigatingCancellationToken);
+            Assert.Equal("IDisposable [from metadata]", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
+            await TestServices.Editor.GoToImplementationAsync(HangMitigatingCancellationToken);
+            var dirtyModifier = await TestServices.Editor.GetDirtyIndicatorAsync(HangMitigatingCancellationToken);
+            Assert.Equal($"FileImplementation.cs{dirtyModifier}", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
```

Comment:
There isn't a way to check the dirty modifier and base name separately?

Reply:
One could be created